src/MatrixTypes_Native.f90

```diff
@@ -691,11 +692,10 @@ SUBROUTINE assemble_BandedMatrixType(thisMatrix)
   INTEGER(SLK),ALLOCATABLE :: diagRank(:)
 
   REQUIRE(thisMatrix%isInit)
-  REQUIRE(thisMatrix%nnz == thisMatrix%counter)
```

Comment:
Would the requirement be for `thisMatrix%nnz >= thisMatrix%counter` ?

Reply:
Correct, all though this requirement is better placed in the set routine, because otherwise an array access error will happen before the requirement is called (counter is equal to the index of the last set element in iTmp).

I'll make the change and push up.